src/main/java/sx/blah/discord/handle/impl/obj/Category.java

```diff
@@ -0,0 +1,266 @@
+	public final Cache<PermissionOverride> roleOverrides;
+
+	public Category(IShard shard, String name, long id, IGuild guild, int position, boolean isNSFW, Cache<PermissionOverride> userOverrides, Cache<PermissionOverride> roleOverrides) {
+		this.shard = shard;
+		this.client = (DiscordClientImpl) shard.getClient();
+		this.name = name;
+		this.guild = guild;
+		this.position = position;
+		this.id = id;
+		this.isNSFW = isNSFW;
+		this.userOverrides = userOverrides;
+		this.roleOverrides = roleOverrides;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public void setPosition(int position) {
+		this.position = position;
+	}
+
+	public void setNSFW(boolean isNSFW) {
+		this.isNSFW = isNSFW;
+	}
+
+	@Override
+	public void delete() {
+		PermissionUtils.requirePermissions(getModifiedPermissions(getClient().getOurUser()), EnumSet.of(Permissions.MANAGE_CHANNEL));
+
+		((DiscordClientImpl) client).REQUESTS.DELETE.makeRequest(DiscordEndpoints.CHANNELS+id);
+	}
+
+	@Override
+	public boolean isDeleted() {
+		return getClient().getCategoryByID(id) != this;
+	}
+
+	@Override
+	public IGuild getGuild() {
+		return guild;
+	}
+
+	@Override
+	public int getPosition() {
+		return position;
+	}
+
+	@Override
+	public void changePosition(int position) {
+		edit(new ChannelEditRequest.Builder().position(position).build());
+	}
+
+	private void edit(ChannelEditRequest request) {
+		PermissionUtils.requirePermissions(getModifiedPermissions(client.getOurUser()), EnumSet.of(Permissions.MANAGE_CHANNEL, Permissions.MANAGE_CHANNELS));
+
+		try {
+			client.REQUESTS.PATCH.makeRequest(
+					DiscordEndpoints.CHANNELS + id,
+					DiscordUtils.MAPPER.writeValueAsString(request));
+		} catch (JsonProcessingException e) {
+			Discord4J.LOGGER.error(LogMarkers.HANDLE, "Discord4J Internal Exception", e);
+		}
+	}
+
+	@Override
+	public String getName() {
+		return name;
+	}
+
+	@Override
+	public void changeName(String name) {
+		if (name == null || !name.matches("^[a-z0-9-_]{2,100}$"))
+			throw new IllegalArgumentException("Channel name must be 2-100 alphanumeric characters.");
+
+		edit(new ChannelEditRequest.Builder().name(name).build());
+	}
+
+	@Override
+	public IDiscordClient getClient() {
+		return client;
+	}
+
+	@Override
+	public IShard getShard() {
+		return shard;
+	}
+
+	@Override
+	public boolean isNSFW() {
+		return isNSFW;
+	}
+
+	@Override
+	public void changeNSFW(boolean nsfw) {
+		edit(new ChannelEditRequest.Builder().nsfw(isNSFW).build());
+	}
+
+	@Override
+	public ICategory copy() {
+		return new Category(shard, name, id, guild, position, isNSFW, userOverrides, roleOverrides);
+	}
+
+	@Override
+	public long getLongID() {
+		return id;
+	}
+
+	@Override
+	public List<IChannel> getChannels() {
+		return getGuild().getChannels().stream()
+				.filter(channel -> channel.getCategory().equals(this))
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public EnumSet<Permissions> getModifiedPermissions(IUser user) {
```

Comment:
Code duplication with the Channel object, this should be moved to a static helper method.